angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -0,0 +1,142 @@
+import { Component, OnInit, HostBinding, Input, Output, EventEmitter, ViewContainerRef, TemplateRef, ViewChild, ElementRef } from '@angular/core';
+import { TimePickerService, MinuteIntervalType } from './time-picker.service';
+import {
+  Overlay,
+  OverlayConfig,
+  OverlayRef
+} from '@angular/cdk/overlay';
+import { TemplatePortal } from '@angular/cdk/portal';
+export declare type HorizontalConnectionPos = 'start' | 'center' | 'end';
+export declare type VerticalConnectionPos = 'top' | 'center' | 'bottom';
+
+@Component({
+  selector: 'cui-time-picker',
+  template: `
+    <div class='cui-input-group  read-only' #connectToInput>
+      <input class="cui-input read-only dirty" 
+      tabindex="0" 
+      type="text" 
+      name="cui-timepicker__input-407" 
+      readonly=""
+      (focus) = 'onFocus()'
+      [value]="value">
+    </div>
+    <ng-template #timePickerCon>
+      <cui-time-picker-dropdown>
+        <cui-time-selector unit='h'></cui-time-selector>:
+        <cui-time-selector unit='m'></cui-time-selector>
+        <cui-time-selector unit='pre' *ngIf='!militaryTime'></cui-time-selector>
+      </cui-time-picker-dropdown>
+    </ng-template>
+  `,
+  styles: [],
+  providers:[TimePickerService]
+})
+export class TimePickerComponent implements OnInit {
+
+  private overlayRef: OverlayRef;
+  private tp: TemplatePortal;
+
+  /** @prop Set the value of the Input element | '' */
+  @Input() value:string = '';
+  /** @prop Optional CSS class name | '' */
+  @Input() className:string = '';
+  /** @prop Set Input element ID | '' */
+  @Input() inputId:string ='';
+  /** @prop Choose to use military time | false */
+  @Input() militaryTime:boolean = false;
+  /** @prop Determine the minute interval | 1 */
+  @Input() minuteInterval:MinuteIntervalType =1;
+  /** @prop Set the initial selected time | null */
+  @Input() selectedTime:any;
+  /** @prop Optional overlay positioin | '' */
+  @Input() public originX: HorizontalConnectionPos = 'start';
+  /** @prop Optional overlay positioin | '' */
+  @Input() public originY: VerticalConnectionPos = 'bottom';
+  /** @prop Optional overlay positioin | '' */
+  @Input() public overlayX: HorizontalConnectionPos ='start';
+  /** @prop Optional overlay positioin | '' */
+  @Input() public overlayY: VerticalConnectionPos = 'top';
+  /** @prop Callback function invoked when user makes a change | null */
+  @Output() whenChange = new EventEmitter();
+
+  @HostBinding('class') classes = 'cui-timepicker-container';
+
+  @ViewChild('timePickerCon') timePickerCon: TemplateRef<any>;
+  @ViewChild('connectToInput') connectToInput: ElementRef;
+
+  constructor(
+    public timePickerService:TimePickerService,
```

Comment:
why are you using `let ts`, if you wanted, you could have used ts here
```suggestion
    public ts: TimePickerService,
```

Reply:
this.ts will do a hash search operation in cash, just store the address of the variable to save time and make the code shorter.